### PR TITLE
Thief pacifist

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR

add pacified to thief 

## Why / Balance

I really don't want to add pacifism to the thief, but mikucult feel it's necessary.

:cl:
- tweak: Thieves are now pacifists.

### CODE REVIEW

**Comment by deltanedas** [comment]
pacifist :trollface: pacific is an ocean

File: Content.Server/GameTicking/Rules/ThiefRuleSystem.cs [hunk 1]
```diff
@@ -123,6 +124,12 @@ public bool MakeThief(ICommonSession thief)
             PrototypeId = thiefRule.ThiefPrototypeId
         });
 
+        //Add Pacified
+        if (!HasComp<PacifiedComponent>(mind.OwnedEntity.Value))
+        {
+            AddComp<PacifiedComponent>(mind.OwnedEntity.Value);
+        }
+
```
Comment (deltanedas) [inline Content.Server/GameTicking/Rules/ThiefRuleSystem.cs]: EnsureComp<PacifiedComponent>(mind.OwnedEntity.Value);

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Hmeister-fake** [comment]
Please do not merge this, you shouldnt remove the ability to defend youself from an antag role

**Comment by Hmeister-fake** [comment]
> Please do not merge this, you shouldnt remove the ability to defend youself from an antag role

Just make them have a check that fails the missions if they kill someone or reach a damage freshhold

File: Content.Server/GameTicking/Rules/Components/ThiefRuleComponent.cs [hunk 2]
```diff
@@ -12,6 +12,11 @@ namespace Content.Server.GameTicking.Rules.Components;
 [RegisterComponent, Access(typeof(ThiefRuleSystem))]
 public sealed partial class ThiefRuleComponent : Component
 {
+    /// <summary>
+    /// Add a Pacified comp to thiefs
+    /// </summary>
+    public bool PacifistThiefs = true;
```
Comment (ficcialfaint) [inline Content.Server/GameTicking/Rules/Components/ThiefRuleComponent.cs:18]:
```suggestion
    /// Add a Pacified comp to thieves
    /// </summary>
    public bool PacifistThieves = true;
```

File: Content.Server/GameTicking/Rules/ThiefRuleSystem.cs [hunk 3]
```diff
@@ -87,11 +88,11 @@ private void DoThiefStart(ThiefRuleComponent component)
 
         foreach(var thief in selectedThieves)
         {
-            MakeThief(thief);
+            MakeThief(thief, component.PacifistThiefs);
```
Comment (ficcialfaint) [inline Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:91]:
```suggestion
            MakeThief(thief, component.PacifistThieves);
```

**Comment by LankLTE** [comment]
Have discussed it in the other PR, but I feel this is a really poor way to stop the antag from attacking people. Rather than discouraging combat, you're outright disallowing it.

File: Content.Server/GameTicking/Rules/Components/ThiefRuleComponent.cs (hunk 2 above)
Comment (deltanedas) [reply Content.Server/GameTicking/Rules/Components/ThiefRuleComponent.cs:18]: Pacify too :trollface:

**Comment by PokkeTigrex** [comment]
My main issue with thieves being pacified over following the rule of no violence/murder is their inability to defend themselves from even the most basic mobs, as well as late round antags like zombies that has the goal to kill everyone, including the thief who needs to survive as well. It also completely rules out breaking in by breaking structures with melee. 
Imagine rolling thief as a salvage, and being unable to do your job because you can't breaks some rocks.

**Comment by LookyPadooky** [comment]
not only is this just kinda whack, but very rarely does anyone ever use the pacifisim trait so if you became a thief most people would just know by the fact that you literally could not throw a punch for your life. maybe call off all thief objectives if the thief directly kills a crewmate? or something like that idk

**Comment by mirrorcult** [comment]
my thoughts from discord but i feel very strongly this is a positive change

image

image

**Comment by mirrorcult** [comment]
the point about rolling salv as a pacifist is valid though. there will probably have to be a proper solution there, maybe just disallowing salv thief

for the record im of the opinion that pacifist trait in general is more fun when the restrictions are fairly obvious and possible to work around--imo it should only block melee attacks and firing guns, pretty much everything else should be allowed

**Comment by LankLTE** [comment]
There's also a possibility sec metagames this by telling prisoners arrested for stealing shit to punch them since if can't punch, then valid. Just banning an role from getting antag because the (kind of arbitrary) restriction it enforces conflicts with that role kind of seems to me like you shouldn't be putting that restriction on it in the first place. 

As a whole if you're going to give them pacifist but then change it to "be possible to work around", you might as well just give them an "inverse objective" to not damage people or kill people too much otherwise they fail which still gives them wiggle room while not making them completely unable to defend themselves from any threat at all.

**Comment by Emisse** [comment]
that doesnt work because random crew mates can be pacifist too

**Comment by LankLTE** [comment]
> that doesnt work because random crew mates can be pacifist too

yeah but it's fairly unlikely, especially in the context of them stealing shit. I doubt the kind of sec who would metagame that are actually gonna be stopped by "well what if I just took the pacifist trait"

**Comment by Aisu9** [comment]
mmmmh.... tasty tasty explosive mix in beakers or chemical grenades, i love pacifism

**Comment by PokkeTigrex** [comment]
Stealing the hypospray (which is very easy with free thieving gloves), or getting the sleep loadout can immediately be weaponized by pacifist thieves to do easy murder. Which is encouraged by forcing this and laxing the rules about non-violence, bypassing the whole point of pacifism.

**Comment by Carmell0** [comment]
This is a twmporary solution, right? Like honestly, there has to be something better. Just make it so the theif does zero damage to humans, or that they only do stun damage.

**Comment by Rainbeon** [comment]
This made dual-antagging Thief-Head Rev very bad for rounds and non-functional for the HR player #22983

**Comment by deltanedas** [comment]
along with not being able to mine, not being able to defend yourself against shitters or ai mobs, sec perma metagaming

this should be reverted

**Comment by Aisu9** [comment]
> along with not being able to mine, not being able to defend yourself against shitters or ai mobs, sec perma metagaming
> 
> this should be reverted

can't you just... use your pickaxe/drill and doafter the blocks? it's slower but still work

**Comment by deltanedas** [comment]
99% sure that was removed
and doafter = thief metagaming

**Comment by mirrorcult** [comment]
making some changes to how pacifist works tomorrow that should alleviate pretty much all issues while keeping the core idea

**Comment by deltanedas** [comment]
that will affect downstream, deltav uses pacifism to prevent end of round grief

**Comment by mirrorcult** [comment]
it shouldnt affect that much at all

**Comment by LankLTE** [comment]
I feel like current pacifist, as a trait, is honestly fine. Changing how pacifist works itself is probably worse than just making some other way of the thief not attack people.

**Comment by Hmeister-fake** [comment]
[#22963] exists for a reason, merge it until a better solution for thief is found.

**Comment by EmoGarbage404** [comment]
> [#22963] exists for a reason, merge it until a better solution for thief is found.

literally just wait a day for the proper solution to be coded. the intent behind the pacifism has been explained thoroughly to you in the discord. you don't need to plug your closed PR because you don't like the design.

**Comment by Hmeister-fake** [comment]
> > [#22963] exists for a reason, merge it until a better solution for thief is found.
> 
> literally just wait a day for the proper solution to be coded. the intent behind the pacifism has been explained thoroughly to you in the discord. you don't need to plug your closed PR because you don't like the design.

It's causing more issues than before due to the new mixed-antag thing. This is quite literally forcing head-revs who also have thief on for example to be pacified and ruining entire shifts. It's just causing more issues than even I thought it would when it first got merged, and its not *just me* not liking the design, im speaking for a lot of people. [#22983] [#22984] and the whole discord discussion with a lot of people disliking it matter, you can't just decide what is best for the game yourselves when there are clear issues.

**Comment by mirrorcult** [comment]
it's Christmas dip shit I already told you I'm fixing it tomorrow

**Comment by Hmeister-fake** [comment]
> it's Christmas dip shit I already told you I'm fixing it tomorrow

Unrelated but fair, forgot timezone wise Americans are in the middle of Christmas cheer rn

**Comment by mirrorcult** [comment]
it's not unrelated it's you can wait a single fuckin day of a feature being mildly broken in like 10% of cases

**Comment by mirrorcult** [comment]
https://github.com/space-wizards/space-station-14/pull/23037

**Pull request merged**